pkg/checker/host_checker.go

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2021 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	v1 "github.com/alibaba/sealer/types/api/v1"
+	v2 "github.com/alibaba/sealer/types/api/v2"
+
+	"github.com/alibaba/sealer/utils/ssh"
+)
+
+type HostChecker struct {
+}
+
+func (a HostChecker) Check(cluster *v2.Cluster, phase string) error {
+	ssh := ssh.NewSSHClient(&cluster.Spec.SSH)
```

Comment:
Each node may has it own ssh config, checkout this: https://github.com/alibaba/sealer/blob/main/pkg/runtime/kubeadm_runtime.go#L94

1. move the function to ssh module,  GetHostSSHClientByIP(hostIP string, *v2.Cluster)
2. The function has bug, it don't merge the ssh config when host ssh config not exist.

Reply:
I will fix those issue.